src/Microsoft.AspNet.Server.Kestrel/Http/FrameRequestStream.cs

```diff
@@ -149,23 +149,27 @@ public void StopAcceptingReads()
         public void Abort()
         {
             // We don't want to throw an ODE until the app func actually completes.
-            // If the request is aborted, we throw an IOException instead.
+            // If the request is aborted, we throw an TaskCanceledException instead.
             if (_state != FrameStreamState.Closed)
             {
                 _state = FrameStreamState.Aborted;
             }
         }
 
-        private void ValidateState()
+        private void ValidateState(CancellationToken cancellationToken)
```

Comment:
Should this return a Task like the `FrameResponseStream` version of `ValidateState` so we can return a faulted task instead of throw synchronously?